addons/idporten.md

```diff
@@ -0,0 +1,339 @@
+
+Make sure to set `aud` value explicit to `issuer` found in [ID-porten well-known]. 
+
+For more details about ID-porten and JWT, see the [JWT grant documentation, same as for Maskinporten].
+
+The final JWT created and send to ID-porten may look like this:
+
+Header, `kid` is keyId retrieved from **`Secrets`** -> `IDPORTEN_CLIENT_JWK`.
+```json
+{
+  "kid": "225ed7ac-33eb-4ce3-bc86-6af40e56868f",
+  "alg": "RS256"
+}
+```
+
+Body, `issuer` in JWT is retrieved from **`Secrets`** -> `IDPORTEN_CLIENT_ID` and `aud` is as described earlier the `issuer` found in [ID-porten well-known]. 
+```json
+{
+  "aud": "https://oidc-ver2.difi.no/idporten-oidc-provider/",
+  "scope": "openid",
+  "iss": "e89006c5-7193-4ca3-8e26-d0990d9d981f",
+  "exp": 1520589928,
+  "iat": 1520589808,
+  "jti": "415ec7ac-33eb-4ce3-bc86-6ad40e29768f"
+}
+```
+
+## Migrating from existing infrastructure-as-code ([IaC]) solution
+
+### Why migrate?
+
+For the same reason as for Azure AD Application:
+
+- Declarative provisioning, straight from your application's [`nais.yaml`](../nais-application/reference.md#spec-idporten)
+- No longer dependent on manual user approvals in multiple IaC repositories
+- No longer dependent on Vault
+- Credentials are rotated on _every_ deploy, completely transparent to the application. 
+This ensures that credentials are fresh and lessens the impact in the case of exposure.
+- The exact same feature is present in the [GCP](../clusters/gcp.md) clusters, 
+which simplifies [migration](../clusters/migrating-to-gcp.md).
+
+### Differences
+
+In general, the ID-porten client provisioned through NAIS are entirely new, unique instances with new client IDs 
+and should be treated as such.
+
+### Migrating - step-by-step
+
+#### 1. Give your application a new description in the [IaC]
+
+In order for NAIS -> [Digdirator] to pick up and update your ID-porten client, the **`description`** of the client registered in 
+the [IaC] should match the expected format:
+
+```
+<cluster>:<namespace>:<app-name>
+```
+
+E.g.
+
+```
+dev-gcp:aura:my-app
+```
+
+Make sure that you are aware of and configure if necessary the value of `spec.idporten.redirectURI`. As redirectURI
+will be generated in this form: `https://my.application.ingress/oauth2/callback`
+
+Take also in consideration to set other values explicit in [`nais.yaml`](../nais-application/reference.md#spec-idporten) 
+as [Digdirator] overrides existing one:  
+`spec.idporten.clientName`  
+`spec.idporten.refreshTokenLifetime`  
+`spec.idporten.clientURI`  
+`spec.idporten.redirectURI`  
+`spec.idporten.frontchannelLogoutURI[]`  
+`spec.idporten.postLogoutRedirectURIs[]`  
+
+## Deletion
+
+The ID-porten client is automatically deleted whenever the associated Application resource is deleted. 
+In other words, if you delete your NAIS application the ID-porten client is also deleted. This will result in
+a **_new and different_** client ID in ID-porten if you re-create the application after deletion.
```

Comment:
Muligheter for å slette en client id i id-porten? Eller er dette kanskje en uuid, så dette er egentlig ikke et problem?

Annet spørsmål, vil man ha behov for å kunne gjenbruke en client id?

Reply:
Med client ID, menes egentlig faktisk klient konfiguration. Så for å ikke få Stakke opp ID-porten med en massa klienter som ikke brukes så er de bra de blir slettet, en klient er ganske tett bundet appen. Men er klienten slettet i maskinporten vill du ikke kunne hente den samme med damme client id. Men en opprettelse av en ny, vill gi deg samme config som førre men ny klient ID og jwk